editor/src/input/input_preprocessor.rs

```diff
@@ -30,13 +32,34 @@ bitflags! {
 	}
 }
 
-#[derive(Debug, Default, Hash)]
+#[derive(Debug, Default)]
 pub struct InputPreprocessor {
 	pub keyboard: KeyStates,
 	pub mouse: MouseState,
 	pub viewport_size: ViewportPosition,
 }
 
+impl InputPreprocessor {
+	/// Returns the byte representation of the message.
+	///
+	/// # Safety
+	/// This function reads from uninitialized memory!!!
+	/// Only use if you know what you are doing
+	unsafe fn as_slice(&self) -> &[u8] {
+		core::slice::from_raw_parts(self as *const InputPreprocessor as *const u8, std::mem::size_of::<Message>())
+	}
+	/// Returns a pseudo hash that should uniquely identify the message.
+	/// This is needed because `Hash` is not implemented for f64s
+	///
+	/// # Safety
+	/// This function reads from uninitialized memory but the generated value should be fine.
+	pub fn pseudo_hash(&self) -> u64 {
+		let mut s = DefaultHasher::new();
+		unsafe { self.as_slice() }.hash(&mut s);
+		s.finish()
+	}
+}
+
```

Comment:
I mean we really don't need unsafe here. I am fine with it but others might disagree

Reply:
Less unsafe is always better. What is needed to fix it?